resources/external_control.urscript

```diff
@@ -126,24 +218,43 @@ thread jointTrajectoryThread():
     exit_critical
     if raw_point[0] > 0:
       q = [ raw_point[1] / MULT_jointstate, raw_point[2] / MULT_jointstate, raw_point[3] / MULT_jointstate, raw_point[4] / MULT_jointstate, raw_point[5] / MULT_jointstate, raw_point[6] / MULT_jointstate]
-      tmptime = raw_point[7] / MULT_time
-      blend_radius = raw_point[8] / MULT_time
-
+      tmptime = raw_point[INDEX_TIME] / MULT_time
+      blend_radius = raw_point[INDEX_BLEND] / MULT_time
       if trajectory_points_left > 0:
-        if raw_point[9] == TRAJECTORY_POINT_JOINT:
-          # textmsg("Executing movej with blending")
+        # MoveJ point
+        if raw_point[INDEX_POINT_TYPE] == TRAJECTORY_POINT_JOINT:
           movej(q, t=tmptime, r=blend_radius)
-        elif raw_point[9] == TRAJECTORY_POINT_CARTESIAN:
-          # textmsg("Executing movel with blending")
+          # reset old acceleration
+          old_qdd = [0, 0, 0, 0, 0, 0]
+        # Movel point
+        elif raw_point[INDEX_POINT_TYPE] == TRAJECTORY_POINT_CARTESIAN:
           movel(p[q[0], q[1], q[2], q[3], q[4], q[5]], t=tmptime, r=blend_radius)
+          old_qdd = [0, 0, 0, 0, 0, 0] # reset old acceleration
+        # Joint spline point
+        elif raw_point[INDEX_POINT_TYPE] == TRAJECTORY_POINT_JOINT_SPLINE:
+          tmptime = raw_point[INDEX_TIME] / MULT_time
+          # Cubic spline
+          if raw_point[INDEX_SPLINE_TYPE] == CUBIC_SPLINES:
+            qd = [ raw_point[7] / MULT_jointstate, raw_point[8] / MULT_jointstate, raw_point[9] / MULT_jointstate, raw_point[10] / MULT_jointstate, raw_point[11] / MULT_jointstate, raw_point[12] / MULT_jointstate]
+            cubicSplineRun(get_actual_joint_positions(), q, get_actual_joint_speeds(), qd, tmptime)
```

Comment:
It should use the target values that have been send to the joint to avoid unnecessary noise 
```suggestion
            cubicSplineRun(get_target_joint_positions(), q, get_target_joint_speeds(), qd, tmptime)
```